inc/util/stream_utils.hpp

```diff
@@ -31,10 +31,12 @@ namespace ebi
 
         // using operator bool: http://www.cplusplus.com/reference/ios/ios/fail/
         // `stream.get()` sets failbit as well on eof
-        do {
-            stream.get(c);
+        while (stream && stream.get(c)) {
             container.push_back(c);
-        } while (stream && c != '\n');
+            if (c == '\n') {
+                break;
+            }
+        }
 
```

Comment:
There were some issues when files ended with different number of newline characters. It was interpreting empty lines as another record, and the parser would go nuts.